Show a list of possible invoice line type (#40)
socialpaymentsbv/docs.clubcollect.com: api-documentation/invoice-line.md

* `paypal`
* `external`
{% endhint %}

## Line types

> The invoice lines can be of 2 basic types: increasing the invoice balance (regular lines, fees, chargebacks) and will have a positive amount attribute or decreasing the invoice balance (payments, credits) and will have a negative amount attribute. Furthermore lines can be categorized by their type (e.g.: Installment fees, Late payment fees).

Reviewer: ```suggestion
The invoice lines can either increase the invoice outstanding (positive amount), e.g. regular lines, fees, chargebacks; or decrease the outstanding (negative amount), e.g. payments, credits. Furthermore lines can be categorised by their type.
```

Reviewer 2: @vruizext `s/whether/either/`

> The type of an invoice line is determined by the `type` attributes (always present) and the optional `subtype` attributes. Below is a list of possible values.

Reviewer: ```suggestion
The type of an invoice line is determined by the `type` attributes (always present) and the optional `subtype` attributes. Below you can find a description of the possible values for invoice lines types.
```

### Regular Lines

This line type is used for lines added to the invoice at creation time, when increasing the invoice amount
from the ClubCollect web interface or when a member adds a donation to the invoice (if enabled).

| Type | Subtype | Description |
|-|-|-|

> | `INVOICE-LINE` | | Regular Invoice line |

Reviewer: Partners don't know what "regular" invoice lines are. Let's try to explain it in their language (and in the same line as in other pages of the API docs)

> Amounts added by the partner when an invoice is created will be of this type

Author: ```suggestion
| `INVOICE-LINE` | | Amounts added by the partner when an invoice is created will be of this type |
```

> | `INVOICE-LINE` | `donation` | Donation Invoice line |

Reviewer: > Donation added to an invoice

Author: ```suggestion
| `INVOICE-LINE` | `donation` | Donation added to an invoice |
```

> | `INVOICE-LINE` | `surcharge` | Payment process fee Invoice line |

Reviewer: > Payment fee charged to the receiver of the invoice

Author: ```suggestion
| `INVOICE-LINE` | `surcharge` | Payment fee charged to the receiver of the invoice |
```

### Fees Lines

At certain points of the life cycle of an invoice fees can be added to the invoice (e.g. Member initiating a payment
in installments).

| Type | Subtype | Description |
|-|-|-|

> | `INSTALLMENT-FEE` | | Fee members must pay when paying for an invoice in installments |

Reviewer: > These fees are added to the invoice when it's paid in installments

Reviewer 2: Prefer `it is` vs. `it’s`.

Author: ```suggestion
| `INSTALLMENT-FEE` | | These fees are added to the invoice when it is paid in installments |
```

> | `CHARGEBACK-FEE` | | When a payment is charged back a fee will be added to the invoice |

Reviewer: > When a payment is charged back a fee might be added to the invoice 

(CBFs aren't always added to the invoice, some clubs opt-out to not charge them, others set a threshold of 1 or 2 chargebacks

Author: ```suggestion
| `CHARGEBACK-FEE` | | When a payment is charged back a fee might be added to the invoice |
```

> | `LATE-PAYMENT-FEE` | | Late payment fee added when reminders are sent |

Reviewer: > Late payment fees are penalties that might be added when the invoice is overdue

Author: ```suggestion
| `LATE-PAYMENT-FEE` | | Late payment fees are penalties that might be added when the invoice is overdue |
```

### Payment Lines

> When payments are made (either directly by member or external payments added from the Club Collect web interface)

Reviewer: I would leave out any mention to "ClubCollect web interface" or to internal logic, trying to keep it simple and straight to the point

> Payment lines are added to register new payments happening for an invoice

Author: ```suggestion
Payment lines are added to register new payments happening for an invoice.
```

payment invoice lines will are added to the invoice.

| Type | Subtype | Description |
|-|-|-|

> | `PAYMENT` | | Payments towards the regular invoice lines |

Reviewer: > Payment for the amount invoiced by the partner

Author: ```suggestion
| `PAYMENT` | | Payment for the amount invoiced by the partner |
```

| `PAYMENT` | `donation` | Payments towards donation invoice lines |
| `JEUGDFONDS` | `jsf` | Jeugdfonds payment |
| `PAYMENT-INSTALLMENT-FEE` | | Payments for installment fees |
| `PAYMENT-CHARGEBACK-FEE` | | Payments for chargeback fees |
| `PAYMENT-LATE-PAYMENT-FEE` | | Payments for late payment fees |
| `PAYMENT-PENALTY-FEE` | | Payment for penalty fees (Legacy) |

## Chargeback Lines

When a previously successful payment is charged back (usually happens for Sepa Direct Direct or Card payments)

> Club Collect will generate chargeback lines for the registered payment lines.

Reviewer: I would reword this:

> When a successful payment (usually Sepa Direct Debit) is charged back, chargeback lines are added to the invoice to cancel the amount of the payment that has been charged back.

> When a previously successful payment is charged back (usually happens for Sepa Direct Direct or Card payments)
> Club Collect will generate chargeback lines for the registered payment lines.

Author: ```suggestion
When a successful payment (usually Sepa Direct Debit) is charged back, chargeback lines are added to the invoice to cancel the amount of the payment that has been charged back.
```

| Type | Subtype | Description |
|-|-|-|

> | `CHARGEBACK` | | Chargeback line for regular invoice lines |

Reviewer: > Chargeback for the amount invoiced by the partner

Reviewer 2: Technically the amount is invoiced by the partner. “Legally” the amount is invoiced by the club (via the partner). Not sure if having “by the partner” will clarify or cloud the meaning. The alternatives:

- nothing (`Chargeback`)
- explicit (`Chargeback for the amount invoiced by the partner on behalf of the club`)
- current suggestion (`Chargeback for the amount invoiced by the partner`)

Another technical note, is this not the chargeback of a _payment_ for the amount invoiced, rather than a chargeback for the _amount invoiced_. ?

Author: ```suggestion
| `CHARGEBACK` | | Chargeback for the amount invoiced by the partner |
```

Reviewer: @pootsbook I used "by the partner" because we use it in other pages in the API docs, IIRC for invoices, we say that "the partner" does this or that.

| `CHARGEBACK` | `donation` | Chargeback line for donation invoice lines |
| `CHARGEBACK-INSTALLMENT-FEE` | | Chargeback for installment fees |
| `CHARGEBACK-LATE-PAYMENT-FEE` | | Chargeback for late payment fee |
| `CHARGEBACK-CHARGEBACK-FEE` | | Chargeback for paid chargeback fees |
| `CHARGEBACK-PENALTY-FEE` | | Chargeback for penalty fees |

## Credit Lines

> When credits are added to an invocie or fees are credited from the Club Collect Web Interface we will add

Reviewer: I'm not convinced about this wording, it's a bit overcomplicated and giving some details that are unnecessary. Let's keep it simple 

```suggestion
Credit lines are added to the invoice to discount an amount from the amount outstanding. 
```

a new invoice line to the invoice.

| Type | Subtype | Description |
|-|-|-|

> | `CREDIT-LINE` | | General credit line |

Reviewer: We don't talk of "General" credit lines anywhere. 

I would say, trying to keep the vocabulary / wording consistent:

> Discount applied to the amount invoiced by the partner

Author: ```suggestion
| `CREDIT-LINE` | | Discount applied to the amount invoiced by the partner |
```

| `CREDIT-LINE` | `donation` | Credit for donation lines |
| `CREDIT-LINE` | `surcharge` | Credits for the payment fee |
| `CREDIT-CHARGEBACK-FEE` | | Credits for a chargeback fee |
| `CREDIT-INSTALLMENT-FEE` | | Credit for an installment fee |
| `CREDIT-LATE-PAYMENT-FEE` | | Credit for a late payment fee |
| `CREDIT-PENALTY-FEE` | | Credit for a Penalty fee |

> Note: penalty lines are a legacy types. You'll see this in the API for older invoices but no new invoice lines

Reviewer: ```suggestion
Note: penalty lines are a legacy type. You might see some of them for older invoices but no new invoice lines of this type are created.
```

Reviewer 2: `s/are a legacy types./are legacy types./` OR
`s/are a legacy types./are a legacy type./`

| Type | Subtype | Description |
|-|-|-|

> | `INVOICE-LINE` | | Amounts added by the partner when an invoice is created will be of this type |

Reviewer: @cristianbica there is an extra ` | ` here?

Reviewer: nvm, it's an empty cell
